src/connectFactoryObservable.ts

```diff
@@ -37,32 +38,29 @@ export function connectFactoryObservable<
   (...args: A) => Exclude<O, typeof SUSPENSE>,
   (...args: A) => Observable<O>,
 ] {
-  const cache = new Map<
-    string,
-    [BehaviorObservable<O>, BehaviorObservable<O>]
-  >()
+  const cache = new Map<string, [Observable<O>, BehaviorObservable<O>]>()
 
   const getSharedObservables$ = (
     ...input: A
-  ): [BehaviorObservable<O>, BehaviorObservable<O>] => {
+  ): [Observable<O>, BehaviorObservable<O>] => {
     const key = JSON.stringify(input)
     const cachedVal = cache.get(key)
 
     if (cachedVal !== undefined) {
       return cachedVal
     }
 
-    const sharedObservable$ = shareLatest<O>(false, () => {
+    const sharedObservable$ = shareLatest(getObservable(...input), () => {
       cache.delete(key)
-    })(getObservable(...input))
+    })
 
     const reactObservable$ = reactEnhancer(
       sharedObservable$,
       unsubscribeGraceTime,
     )
 
-    const result: [BehaviorObservable<O>, BehaviorObservable<O>] = [
-      sharedObservable$,
+    const result: [Observable<O>, BehaviorObservable<O>] = [
+      takeUntilComplete(sharedObservable$),
       reactObservable$,
     ]
```

Comment:
It's a bit unrelated to this PR, but I always found confusing the fact that the cache holds `[sharedObservable, reactObservable]` and this method returns `[reactHook, sharedObservable]` 😅

Reply:
Dah! This is a very good point, I will address that in a separate commit, for sure. Thanks for pointing that out @voliva